server/routes/api.md

```diff
@@ -0,0 +1,129 @@
+# API
+
+* Routes
+  * [POST /api/users](#post-apiusers)
+  * [GET /api/users/:handle](#get-apiusershandle)
+  * [POST /api/users/:handle/o-auth-identities](#post-apiusershandleo-auth-identities)
+  * [GET /auth/login-o-auth](#get-authlogin-o-auth)
+* Resources
+  * [Users](#users)
+  
+## Basics
+* Examples are in JavaScript on a Node/Express server with [request](https://github.com/request/request) installed.
+* Request and responses are in JSON.
+* API responses are the base resource being created/referenced. So, for example, all routes starting with `/api/users` return [User](#users) resources.
+
+## Client Authentication
+
+API routes must be called with Basic HTTP Authentication. Provide your username and password with each request.
+
+```javascript
+url = 'https://codecombat.com/api/users'
+json = { name: 'A username' }
+auth = { name: CLIENT_ID, pass: CLIENT_SECRET }
+request.get({ url, json, auth }, (err, res) => console.log(res.statusCode, res.body))
+```
+
+## User Authentication
+
+To authenticate a user on CodeCombat through your service, you will need to use OAuth 2. CodeCombat will act as the client, and your service will act as the provider. Your service will need to provide a trusted lookup URL where we can send the tokens given to us by users and receive user information. The process from user account creation to log in will look like this:
+
+1. **Create the user**: Use `POST /api/users`.
+1. **Link the CodeCombat user to an OAuth identity**: Use `POST /api/users/:handle/oauth-identities`, providing a token and your OAuth id which we will generate for you.
```

Comment:
That's described in detail below; this is just an outline of the process. Maybe link it? How about instead of touching on the specific parameters, changing it to "use `POST /api/users/:handle/oauth-identities`, to link a given CodeCombat user to an account on your own service."?